### PR TITLE
Fix OpenApi spec - add missing definitions

### PR DESCRIPTION
**Pull request by ghost**
### 1. Why is this change necessary?
When accessing the generated OpenApi spec of a Shopware 6 shop, parts of the spec are missing.
The 'error' specification is missing the definition of the 'description' field, which leads to errors.
Also, the 'responses' section was missing from the 'components' section of the spec, although the file for it exists and is loaded. This missing 'responses' section results in an invalid OpenApi spec.
There are a few type and pattern defintions missing from the OpenApi json files.

The provided changes fix all the bugs and make the OpenApi spec valid again.

### 2. What does this change do, exactly?
- Adds the missing 'description' definition to the 'error' schema in the OpenApiSchemaBuilder
- Adds the missing 'responses' section in the 'components' section in the OpenApi Spec by a update to the array merge, to include not just the schema-components but also the responses
- Adds the missing type and pattern defintions to a few entries of the OpenApi json files

### 3. Describe each step to reproduce the issue or behaviour.
- Create a new Shopware 6.5 project (with the production template)
- Open the OpenApi Spec (<SHOP_DOMAIN>/store-api/_info/openapi3.json)

### 5. Checklist

- [x] I have rebased my changes to remove merge conflicts
- [x] I have written tests and verified that they fail without my change
- [X] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfil them.

### <samp>🤖 Generated by Copilot at 2a3ffa2</samp>

This pull request improves the OpenApi core specification for the shopware/platform API by adding responses, descriptions and components. It also updates the changelog and removes unused code.

### <samp>🤖 Generated by Copilot at 2a3ffa2</samp>

*  Add a changelog file to document the pull request ([link](https://github.com/shopware/platform/pull/3182/files?diff=unified&w=0#diff-adcbad9324e43b5643e03e671234f16dfb4b5039dd801e2b01068cffb9fd47caR1-R10))
*  Add a description property to the error schema in the OpenApiSchemaBuilder class to provide a human-readable explanation of the error ([link](https://github.com/shopware/platform/pull/3182/files?diff=unified&w=0#diff-174b52f8037815ce2cc100906b96923b7a5dc763120ec9cad3302ee4c0361f54R355))
*  Add a responses section to the OpenApi specification in the `OpenApiFileLoader.php` file to define the possible responses for each API endpoint ([link](https://github.com/shopware/platform/pull/3182/files?diff=unified&w=0#diff-c54e695e7ad55305568cdf6725c92cd338540a16f76d18fc68c4d1db103770a6R32))
*  Modify the components merging logic in the `OpenApiFileLoader.php` file to merge the entire components section recursively instead of only the schemas sub-section ([link](https://github.com/shopware/platform/pull/3182/files?diff=unified&w=0#diff-c54e695e7ad55305568cdf6725c92cd338540a16f76d18fc68c4d1db103770a6L47-R49))
*  Remove an unused use statement from the `OpenApiFileLoader.php` file to optimize the code ([link](https://github.com/shopware/platform/pull/3182/files?diff=unified&w=0#diff-c54e695e7ad55305568cdf6725c92cd338540a16f76d18fc68c4d1db103770a6L5))

### CODE REVIEW

**Comment by ghost** [comment]
Hello,

thank you for creating this pull request.
I have opened an issue on our Issue Tracker for you. See the issue link: https://issues.shopware.com/issues/NEXT-28688

Please use this issue to track the state of your pull request.

**Comment by ghost** [comment]
Adjusted the test class & Added even more missing types & patterns back to the path files

**Comment by akf-bw** [comment]
Added even more missing types & patterns back to the path files

File: src/Core/Framework/Api/ApiDefinition/DefinitionService.php [hunk 1]
```diff
@@ -11,7 +11,7 @@
 /**
  * @phpstan-type Api DefinitionService::API|DefinitionService::STORE_API
  * @phpstan-type ApiType DefinitionService::TypeJsonApi|DefinitionService::TypeJson
- * @phpstan-type OpenApiSpec  array{paths: array<string,array<mixed>>, components: array{schemas: array<string, array<mixed>>}}
+ * @phpstan-type OpenApiSpec  array{paths: array<string,array<mixed>>, components: array<mixed>}
```
Comment (aragon999) [inline src/Core/Framework/Api/ApiDefinition/DefinitionService.php]:
```suggestion
 * @phpstan-type OpenApiSpec array{paths: array<string,array<mixed>>, components: array<mixed>}
```

**Comment by ghost** [comment]
Rebased all trunk branch changes and squashed all changes into a single commit. @Christian-Rades this can now be reviewed/merged by you.

File: src/Core/Framework/Api/ApiDefinition/DefinitionService.php (hunk 1 above)
Comment (ghost) [reply src/Core/Framework/Api/ApiDefinition/DefinitionService.php]: Accepted and added

**Comment by Christian-Rades** [comment]
Thank you for your contribution :tada: .
Sorry about the wait, it looks good to me. The PR will be merged shortly.

**Comment by shyim** [comment]
Thanks!

**Pull request closed without merging**